src/main/java/io/r2dbc/pool/ConnectionPool.java

```diff
@@ -163,27 +164,50 @@ public void close() {
         dispose();
     }
 
+    /**
+     * Dispose this {@link ConnectionPool} in non-blocking flow.
+     * <p>
+     * When multiple errors occurred during dispose flow, they are added as
+     * suppressed errors onto the first error.
+     *
+     * @return a Mono triggering the shutdown of the pool once subscribed.
+     */
+    public Mono<Void> disposeLater() {
+        List<Throwable> errors = new ArrayList<>();
+        return Flux.fromIterable(this.destroyHandlers)
+            .flatMap(Mono::fromRunnable)
+            .concatWith(this.connectionPool.disposeLater())
+            .onErrorContinue((throwable, o) -> {
+                errors.add(throwable);
+            })
+            .then(Mono.defer(() -> {
+                if (errors.isEmpty()) {
+                    return Mono.empty();
+                }
+
+                Throwable rootError = errors.get(0);
+                if (errors.size() == 1) {
+                    return Mono.error(rootError);
+                }
+
+                errors.subList(1, errors.size()).forEach(rootError::addSuppressed);
+
+                return Mono.error(rootError);
+            }));
+    }
+
```

Comment:
Let's add the blocking behavior for now into the `close()` method (`disposeLater().block()`) to progress with the PR until SPI gets the `Closeable` spec merged.

Reply:
So, you mean change `close()` method implementation to:

```java
@Override
public void close() {
    disposeLater().block();
}
```

Reply:
Yes, that's the idea for now.